### PR TITLE
fix: use original CSS source for declaration source map positions

### PR DESCRIPTION
**Pull request by brettburley**
## Problem

Declaration source maps (used for "Go to Definition" from `.d.ts` to CSS) used `inputCss` from the Vite transform hook, which may already be processed by PostCSS plugins. This caused source map positions to reflect the **transformed** CSS rather than the **original** source file.

Whether PostCSS processes the CSS before our transform hook is non-deterministic — it depends on Vite resolving the PostCSS config relative to `process.cwd()`, which varies in monorepo build systems like Nx.

## Changes

- Cache original CSS in the `enforce: "pre"` transform hook on `patch-css-modules` (runs before `vite:css` applies PostCSS). Pass the cache to the inner plugin for use in source map generation.
- Replace PostCSS + postcss-selector-parser with a character-by-character text scanner (`find-class-positions.ts`) that skips comments and strings. Works across all CSS-like syntaxes (CSS, SCSS, Less, Sass indented, Stylus) where PostCSS fails on Sass and Stylus.
- Handle CSS-escaped class names (e.g. `.foo\:bar`) and skip Less mixin calls (`.name()`).
- Treat non-ASCII characters (codepoint >= 128) as valid CSS identifier characters.
- Add Vite hook filter on the pre-transform to skip non-CSS files.
- Use `startsWith` instead of `slice` for zero-allocation string comparison in the scanner.
- Add 26 unit tests for the scanner covering all edge cases.
- Add integration tests for both PostCSS and LightningCSS declarationMap code paths.

<details>
<summary>Original PR description</summary>

## Problem

`findClassPositions` parses the `inputCss` parameter from the Vite transform hook to compute source map positions in generated `.module.css.d.ts` files. However, `inputCss` may already be processed by PostCSS plugins (e.g. `postcss-nested`), causing positions to reflect the **transformed** CSS rather than the **original** source file.

For example, with `postcss-nested`, a nested selector like `&.nested` (line 4, col 3 in the original) gets flattened to `.parent.nested` (line 2, col 8 in the output). The source map claims to point to the original `.module.css` file, but the positions come from the post-transform CSS.

### Why this is non-deterministic

Whether PostCSS processes the CSS before it reaches the `vite-css-modules` transform hook depends on Vite's `compilePostCSS` resolving the postcss config. The config resolution uses `path.resolve(css.postcss)` relative to `process.cwd()`. In monorepo build systems like Nx, CWD varies between Vite's internal phases within a single build:

```
process.cwd() = /repo-root              (config resolution phase)
process.cwd() = /repo-root/libs/my-lib  (build phase)
```

Since `patchCssModules` sets `config.css.modules = false`, Vite's `compilePostCSS` early-returns when it can't find the postcss config (`if (!postcssConfig && !isModule && !needInlineImport && !hasUrl) return`). This means `inputCss` is sometimes raw nested CSS and sometimes flattened CSS — producing different source maps on each build.

## Solution

`findClassPositions` now reads the original CSS from disk (`readFile(filePath, "utf8")`) instead of accepting the potentially-transformed `inputCss`. The `filePath` parameter was already available and validated. This ensures source map positions always correspond to the original source file.

## Test

Added a test that uses `postcss-nested` to flatten nested CSS and verifies source map positions point to the **original** nested positions (e.g. `&.nested` at line 4), not the flattened positions.

</details>

### CODE REVIEW

File: src/plugin/find-class-positions.ts [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+			line += 1;
+			column = 1;
+		} else {
+			column += 1;
+		}
+		i += 1;
+	}
+
+	return positions;
+};
```
Comment (privatenumber) [inline src/plugin/find-class-positions.ts:377]: Ended up with a lexer so we can support every variation of CSS.
Will likely extract into a separate package later on.
